packages/medusa/src/services/order.ts

```diff
@@ -309,22 +326,63 @@ class OrderService extends TransactionBaseService {
 
   /**
    * Gets an order by id.
-   * @param orderId - id of order to retrieve
+   * @param orderIdOrSelector - id or selector of order to retrieve
    * @param config - config of order to retrieve
    * @return the order document
    */
   async retrieve(
-    orderId: string,
+    orderIdOrSelector: string | Selector<Order>,
```

Comment:
todo: the retrieve method should not allow a selector like this; you can add an internal method if needed otherwise just use `list`?